GoScrumApi/src/main/java/com/goscrum/GoScrumApi/entity/Room.java

```diff
@@ -0,0 +1,84 @@
+package com.GoScrum.GoScrumApi.entity;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.persistence.*;
+import jakarta.validation.constraints.Size;
+import lombok.*;
+
+import java.nio.charset.Charset;
+import java.util.Random;
+import java.util.UUID;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+
+// Database level
+@Entity
+@Table(name = "rooms", uniqueConstraints = {@UniqueConstraint(columnNames = {"slug"})})
+public class Room {
+	@Id
+	@GeneratedValue(strategy = GenerationType.UUID)
+	private UUID id;
+
+	@Column(name = "name", nullable = false)
+	private String name;
+
+	@Column(name = "slug", nullable = false, unique = true)
+	private String slug;
+
+	@Column(name = "description")
+	private String description;
+
+	@Column(name = "is_active", nullable = false)
+	private boolean isActive = true;
+
+	@Column(name = "is_private", nullable = false)
+	private boolean isPrivate;
+
+	@Column(name = "password", nullable = false)
+	private String password;
+
+	@OneToOne
+	@JoinColumn(name = "host_id", nullable = false)
+	private User host;
+
+	@PrePersist
+	private void onPrePersist() {
+		handleDefaultPassword();
```

Comment:
Chuyển qua viết 1 cái method trên service level để tạo room, đừng mang ActiveRecord callback sang JPA. 

Hoặc viết vào `RoomRepository` cũng được nhé.


1. Sample: specifying sql in the repository interface
  
  ```
  
  @Repository
  public interface MerchantRepository extends JpaRepository<MerchantEntity, Integer> {
  
      Optional<MerchantEntity> findByMerchantId(String merchantId);
  
      @Modifying
      @Query("UPDATE MerchantEntity e SET e.status = :status, e.modifiedDate = :#{T(java.time.ZonedDateTime).now()} WHERE e.id = :id")
      void updateStatus(Integer id, String status);
  }
  
  
  ```

1. Sample: implement logic in service class

  ```java
      @Override
      @Transactional
      public MerchantEntity createOrUpdateMerchantIfExist(MerchantInfo merchantInfo) {
          if (merchantInfo == null) {
              throw new IllegalArgumentException();
          }

          Optional<MerchantEntity> merchantEntityOptional = merchantRepository.findByMerchantId(merchantInfo.getMerchantId());
          MerchantEntity merchantEntity;

          if (merchantEntityOptional.isEmpty()) {
              merchantEntity = MerchantEntity.builder()
                      .status(StringUtils.ACTIVE)
                      .build();
          } else {
              merchantEntity = merchantEntityOptional.get();
              merchantEntity.setModifiedDate(ZonedDateTime.now());
          }
  
          return merchantRepository.save(merchantEntity);
      }
  ```